Meta/Lagom/CMakeLists.txt

```diff
@@ -51,281 +79,464 @@ set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${LINKER_FLAGS}")
 
+# FIXME: This is a hack, because the lagom stuff can be built individually or
+#        in combination with the system, we generate two Debug.h files. One in
+#        Build/AK/Debug.h and the other in Build/Meta/Lagom/AK/Debug.h.
+configure_file(../../AK/Debug.h.in AK/Debug.h @ONLY)
+configure_file(../../Kernel/Debug.h.in Kernel/Debug.h @ONLY)
+
 include_directories(../../)
 include_directories(../../Userland/)
 include_directories(../../Userland/Libraries/)
 include_directories(${CMAKE_BINARY_DIR})
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-if (BUILD_LAGOM AND NOT ENABLE_OSS_FUZZ AND NOT ENABLE_FUZZER_SANITIZER)
-    if (CMAKE_SOURCE_DIR MATCHES ".*/Lagom")
-        set(write_if_different ${CMAKE_SOURCE_DIR}/../write-only-on-difference.sh)
-        add_subdirectory(../../Userland/Libraries/LibUnicode/CodeGenerators ${CMAKE_CURRENT_BINARY_DIR}/LibUnicode/CodeGenerators)
-    endif()
-
-    include(../../Userland/Libraries/LibUnicode/unicode_data.cmake)
+# install rules, think about moving to its own helper cmake file
+# Don't install Lagom libs into the target Root/
+# FIXME: Remove this check for 4594
+if (CMAKE_SOURCE_DIR MATCHES ".*/Lagom")
+    include(CMakePackageConfigHelpers)
+    include(GNUInstallDirs)
+
+    # find_package(<package>) call for consumers to find this project
+    set(package Lagom)
+
+    write_basic_package_version_file(
+        "${package}ConfigVersion.cmake"
+        COMPATIBILITY SameMajorVersion
+    )
+
+    # Allow package maintainers to freely override the path for the configs
+    set(Lagom_INSTALL_CMAKEDIR "${CMAKE_INSTALL_DATADIR}/${package}"
+        CACHE PATH "CMake package config location relative to the install prefix")
+    mark_as_advanced(Lagom_INSTALL_CMAKEDIR)
+
+    install(
+        FILES ${SERENITY_PROJECT_ROOT}/Meta/CMake/lagom-install-config.cmake
+        DESTINATION "${Lagom_INSTALL_CMAKEDIR}"
+        RENAME "${package}Config.cmake"
+        COMPONENT Lagom_Development
+    )
+
+    install(
+        FILES "${PROJECT_BINARY_DIR}/${package}ConfigVersion.cmake"
+        DESTINATION "${Lagom_INSTALL_CMAKEDIR}"
+        COMPONENT Lagom_Development
+    )
+
+    install(
+        EXPORT LagomTargets
+        NAMESPACE Lagom::
+        DESTINATION "${Lagom_INSTALL_CMAKEDIR}"
+        COMPONENT Lagom_Development
+    )
+
+    # Manually install AK
+    install(
+        DIRECTORY "${SERENITY_PROJECT_ROOT}/AK"
+        COMPONENT Lagom_Development
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        FILES_MATCHING PATTERN "*.h"
+    )
 endif()
 
+function(lagom_lib library fs_name)
+    cmake_parse_arguments(LAGOM_LIBRARY "" "" "SOURCES;LIBS" ${ARGN})
+    # FIXME: Consider whether to care about -DBUILD_SHARED_LIBS=OFF
+    # Possibly a cmake presets value?
+    set(target_name "Lagom${library}")
+    add_library(${target_name} SHARED ${LAGOM_LIBRARY_SOURCES})
+    # alias for pretty exports
+    add_library(Lagom::${library} ALIAS ${target_name})
```

Comment:
Just a nitpick, but `ALIAS` targets are simply that, targets that alias another one. They do not participate in the install interface of a project.

However, it's wise to make an alias target that is identical to how the target appears in the install interface. This makes it easy to discern how to use a project's targets both when installed or when vendored (e.g. via `FetchContent`). Such an alias is also necessary to make a project ready for vendoring (again, e.g. via `FetchContent`). A fully `FetchContent`-ready project has more considerations as well, but that's outside the scope of this PR from what I can tell.